internal/rbd/replication.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"strings"
+
+	librbd "github.com/ceph/go-ceph/rbd"
+	"github.com/csi-addons/spec/lib/go/replication"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func ResyncVol(localStatus librbd.SiteMirrorImageStatus, rbdVol *rbdVolume, force bool) error {
```

Comment:
this is a great opportunity to cleanup even more. These functions should all be part of the `rbdVolume` object, so make them look like

```go
func (ri *rbdVolume) ResyncVol(localStatus librbd.SiteMirrorImageStatus, force bool) error {
```

These changes can be a 2nd commit in the same PR if you like.

Reply:
This is still open, could you let me know what you want to do with it?

Reply:
Thanks Niels for the review. Yes, addressing all these changes as a part of 2nd commit.  So they are not present in this commit.